### PR TITLE
Authn request extensions

### PR DESCRIPTION
**Pull request by seanski**
This pull request adds the ability to add custom extensions to the AuthnRequest.  I chose to go with adding a setting which takes a lambda to build the extensions since the `samlp:Extensions` attribute allows for a sequence of any element type.  

The `samlp:Extensions` element comes from the `RequestAbstractType`.  More info at http://schemas.stylusstudio.com/saml/nd253446a/complexType_RequestAbstractType.html.

### CODE REVIEW

File: test/request_test.rb [hunk 1]
```diff
@@ -238,5 +238,19 @@ class RequestTest < Test::Unit::TestCase
       auth_doc = OneLogin::RubySaml::Authrequest.new.create_authentication_xml_doc(settings)
       assert auth_doc.to_s =~ /<saml:AuthnContextDeclRef>urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport<\/saml:AuthnContextDeclRef>/
     end
+
+    should 'create the samlp:Extensions element correctly' do
+      settings = OneLogin::RubySaml::Settings.new
+      settings.idp_sso_target_url = 'http://example.com'
+      settings.authn_request_extensions_builder = ->(extension_context) {
```
Comment (Lordnibbler) [inline test/request_test.rb]: can you adjust this to use regular block syntax for 1.8.x compatibility

**Comment by Lordnibbler** [comment]
You'll need to rebase this branch off `master` and update your tests to use Minitest please! :)

**Comment by Lordnibbler** [comment]
@seanski are you still planning to update this PR?

**Comment by seanski** [comment]
@Lordnibbler - Sorry for the delay.  I would love to take care of these, and I can do it within the next couple of days.

**Comment by levelboy** [comment]
@seanski @Lordnibbler Do you still plan to merge this PR?

**Pull request closed without merging**